pkg/virt-controller/watch/export/export.go

```diff
@@ -507,11 +570,94 @@ func (ctrl *VMExportController) handleIsVmSnapshot(vmExport *exportv1.VirtualMac
 	return 0, ctrl.updateVMExporVMSnapshotStatus(vmExport, pvcs, pod, service, false)
 }
 
+func (ctrl *VMExportController) handleIsVM(vmExport *exportv1.VirtualMachineExport, service *corev1.Service) (time.Duration, error) {
+	var pvcs []*corev1.PersistentVolumeClaim
+
+	inUse := false
+	vmi, exists, err := ctrl.getVmi(vmExport.Namespace, vmExport.Spec.Source.Name)
+	if err != nil {
+		return 0, err
+	}
+	if exists {
+		inUse = true
+	}
+
+	pod, exists, err := ctrl.getExporterPod(vmExport)
+	if err != nil {
+		return 0, err
+	}
+	pvcs, err = ctrl.getPVCsFromVM(vmExport.Namespace, vmExport.Spec.Source.Name)
+	if err != nil {
+		return 0, err
+	}
+	if !exists {
+		if !inUse {
+			pod, err = ctrl.createPodAndSecret(vmExport, pvcs)
+			if err != nil {
+				return 0, err
+			}
+		}
+	} else {
+		if inUse {
+			// VMI exists, stop pod if it exists.
+			if err := ctrl.deleteExporterPod(vmExport, pod, vmStartedEvent, fmt.Sprintf("virtual machine %s/%s was started", vmi.Namespace, vmi.Name)); err != nil {
+				return 0, err
+			}
+			pod = nil
+		} else {
+			if err := ctrl.handlePodSucceededOrFailed(vmExport, pod); err != nil {
+				return 0, err
+			}
+		}
+	}
+
+	return 0, ctrl.updateVMExporVMStatus(vmExport, pvcs, pod, service, inUse)
+}
+
+func (ctrl *VMExportController) getPVCsFromVM(vmNamespace, vmName string) ([]*corev1.PersistentVolumeClaim, error) {
+	var pvcs []*corev1.PersistentVolumeClaim
+	vm, exists, err := ctrl.getVm(vmNamespace, vmName)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, nil
+	}
+	name := ""
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		if volume.PersistentVolumeClaim != nil {
+			name = volume.PersistentVolumeClaim.ClaimName
+		}
+		if volume.DataVolume != nil {
+			name = volume.DataVolume.Name
+		}
+		if volume.MemoryDump != nil {
+			name = volume.MemoryDump.ClaimName
+		}
+		pvc, exists, err := ctrl.getPvc(vmNamespace, name)
+		if err != nil {
+			return nil, nil
+		}
+		if exists {
+			pvcs = append(pvcs, pvc)
+		}
+		name = ""
```

Comment:
Why not just declare this variable in the loop?

Reply:
Don't know changed.